File: python/Exercises/Factors.py
def factors(n):
    '''check input number's factor'''
    factor = 1
    while factor < (n / factor):
        factor = factor + 1
        while (n % factor == 0):
            n = n / factor
            print(factor)
    if n > 1:
        print(n)

factors(1248)

File: java/Factors.java
public class Factors {
    public static void main(String[] args) {
        //input variable
        long n = Long.parseLong(args[0]);

        for (long factor = 2; factor <= n / factor; factor++)
        {
            while (n % factor == 0)
            {
                n /= factor;
                System.out.println(factor + " ");
            }
        }
        if (n > 1) System.out.print(n);
        System.out.println();

    
    }
}
